cairo/src/context.rs

```diff
@@ -435,19 +445,22 @@ impl Context {
         unsafe {
             ffi::cairo_stroke_extents(self.0.as_ptr(), &mut x1, &mut y1, &mut x2, &mut y2);
         }
-        (x1, y1, x2, y2)
+        self.status().map(|_| (x1, y1, x2, y2))
     }
 
-    pub fn in_stroke(&self, x: f64, y: f64) -> bool {
-        unsafe { ffi::cairo_in_stroke(self.0.as_ptr(), x, y).as_bool() }
+    pub fn in_stroke(&self, x: f64, y: f64) -> Result<bool, Error> {
+        let in_stroke = unsafe { ffi::cairo_in_stroke(self.0.as_ptr(), x, y).as_bool() };
+        self.status().map(|_| in_stroke)
     }
 
-    pub fn copy_page(&self) {
-        unsafe { ffi::cairo_copy_page(self.0.as_ptr()) }
+    pub fn copy_page(&self) -> Result<(), Error> {
+        unsafe { ffi::cairo_copy_page(self.0.as_ptr()) };
+        self.status()
     }
 
-    pub fn show_page(&self) {
-        unsafe { ffi::cairo_show_page(self.0.as_ptr()) }
+    pub fn show_page(&self) -> Result<(), Error> {
```

Comment:
And here

Reply:
For PDF/PNG/etc. that are backed by streams, this is where the code gets to know if there was an I/O error (unless you specify a write_func and store an error flag there).